docs/proposals/logdir-volume.md

```diff
@@ -0,0 +1,190 @@
+    ...
+	// LogDir represents a log directory that preserves a pod's log files.
+	LogDir *LogDirVolumeSource `json:"logDir,omitempty"`
+	...
+}
+
+// LogDirVolumeSource represents an log directory for a pod.
+type LogDirVolumeSource struct {
+	// TODO: Support log rotation and maximum log file sizes, etc.
+	// Optional: what type of storage medium should back this directory.
+	// The default is "" which means to use the node's default medium.
+	Medium StorageMedium `json:"medium"`
+}
+```
+
+Add  `logDirPlugin` to implementation `volume.VolumePlugin` interface:
+
+```
+type logDirPlugin struct {
+	host volume.VolumeHost
+}
+
+const (
+	logDirPluginName = "kubernetes.io/log-dir"
+)
+
+func (plugin *logDirPlugin) Init(host volume.VolumeHost) {
+	plugin.host = host
+}
+
+func (plugin *logDirPlugin) Name() string {
+	return logDirPluginName
+}
+
+func (plugin *logDirPlugin) CanSupport(spec *volume.Spec) bool {
+	if spec.VolumeSource.LogDir != nil {
+		return true
+	}
+	return false
+}
+
+func (plugin *logDirPlugin) NewBuilder(spec *volume.Spec, pod *api.Pod, opts volume.VolumeOptions, mounter mount.Interface) (volume.Builder, error) {
+	return plugin.newBuilderInternal(spec, pod, mounter, &realMountDetector{mounter}, opts, newChconRunner())
+}
+
+func (plugin *logDirPlugin) NewCleaner(volName string, podUID types.UID, mounter mount.Interface) (volume.Cleaner, error) {
+	// Inject real implementations here, test through the internal function.
+	return plugin.newCleanerInternal(volName, podUID, mounter, &realMountDetector{mounter})
+}
+```
+
+## Integrated with Fluentd
+
+We can use Fluentd to collect log files in `LogDir` volume. Fluentd should be installed on each Kubernetes node, and it will watch `LogDir` volume's path, the Fluentd's configuration as follows(we use kubelet_root_dir's defalut value /var/lib/kubelet, ):
+
+```
+<source>
+  type tail
+  format none
+  time_key time
+  path /var/lib/kubelet/pods/*/volumes/kubernetes.io~log-dir/**/*.log
+  pos_file /lib/pods.log.pos
+  time_format %Y-%m-%dT%H:%M:%S
+  tag reform.*
+  read_from_head true
+</source>
+
+<match reform.**>
+  type record_reformer
+  enable_ruby true
+  tag kubernetes.${hostname}.${tag_suffix[10]}
+</match>
+
+<match **>
+  type stdout
+</match>
+```
+
+The Fluentd will tail any files in the `LogDir` volume, and add tag `kubernetes.<node_host_name>.<pod_full_name>.<file_name>`. We only print the logs to stdout,  also can forward to logging storage endpoint(e.g Elasticsearch).
+
+For example, we have a application writes log files:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: my-app
+spec:
+  containers:
+  - name: container1
+    image: ubuntu:14.04
+    command: ["bash", "-c", "i=\"0\"; while true; do echo \"`hostname`_container1: $i \"; date --rfc-3339 ns >> /varlog/container1.log; sleep 4; i=$[$i+1]; done"]
+    volumeMounts:
+    - name: log-storage
+      mountPath: /varlog
+    securityContext:
+        privileged: true
+  volumes:
+  - name: log-storage
```

Comment:
In the case that I want to collect two different paths, am I expected to create two such volumes, or mount the same volume on two different paths?